hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/CacheWarmer.java

```diff
@@ -59,13 +60,23 @@ public CacheWarmer(
             @NonNull final TransactionChecker checker,
             @NonNull final TransactionDispatcher dispatcher,
             @NonNull final ConfigProvider configProvider) {
+        this(
+                checker,
+                dispatcher,
+                new ForkJoinPool(configProvider
+                        .getConfiguration()
+                        .getConfigData(CacheConfig.class)
+                        .cryptoTransferWarmThreads()));
+    }
+
+    @VisibleForTesting
+    CacheWarmer(
+            @NonNull final TransactionChecker checker,
+            @NonNull final TransactionDispatcher dispatcher,
+            @NonNull final Executor executor) {
         this.checker = checker;
         this.dispatcher = requireNonNull(dispatcher);
-        final int parallelism = configProvider
-                .getConfiguration()
-                .getConfigData(CacheConfig.class)
-                .cryptoTransferWarmThreads();
-        this.executor = new ForkJoinPool(parallelism);
+        this.executor = requireNonNull(executor);
```

Comment:
Rather than doing it this way, I was wondering if we should use a Dagger provider to create the executor that we use in this constructor. It is more elaborate, but improves the interface. @tinker-michaelj has a good handle on how that would work.

Reply:
Yes, makes sense. In particular, once the `ExecutorFactory` from platform-base is available, this will be much cleaner.